python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -127,15 +127,29 @@ def broadcast(*columns: Column, target_length: int | None = None) -> list[Column
     ]
 
 
+class PartitionInfo:
+    """
+    Partitioning information.
+
+    This class only tracks the partition count (for now).
+    """
+
+    __slots__ = ("count",)
+
+    def __init__(self, count: int):
+        self.count = count
```

Comment:
This class is very simple in this PR. I expect we will also want to keep track of which column(s) we are hash-partitioned on in the future.